### PR TITLE
Add SIG Testing Spotlight article

### PR DESCRIPTION
**Pull request by fsmunoz**
Add SIG Testing spotlight article

Closes: #413

### CODE REVIEW

File: content/en/blog/2023/sig-testing-spotlight.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+as well as boskos (resource rental for e2e tests),
+[KIND](https://kind.sigs.k8s.io/) (Kubernetes-in-Docker, for local
+testing and development), and the cloud provider for KIND.  For the
+latter, there's [Prow](https://docs.prow.k8s.io/) (K8s-based CI/CD and
+chatops), and a litany of other tools and utilities for triage,
+analysis, coverage, Prow/TestGrid config generation, and more in the
+test-infra repo.
+
+*If you are willing to learn more and get involved with any of the SIG
+Testing sub-projects, check out the [SIG Testing README](https://github.com/kubernetes/community/tree/master/sig-testing#subprojects).*
```
Comment (sandipanpanda) [inline content/en/blog/2023/sig-testing-spotlight.md:87]: Thank you @fsmunoz! A nit:
```suggestion
Testing subprojects, check out the [SIG Testing README](https://github.com/kubernetes/community/tree/master/sig-testing#subprojects).*
```

Comment (fsmunoz) [reply content/en/blog/2023/sig-testing-spotlight.md:87]: Yes, we have both present in the documentation and blogs, but subprojects is likely better (and it's your text to begin with) so I'll add it.

File: content/en/blog/2023/sig-testing-spotlight.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
+---
+layout: blog
+title: "Spotlight on SIG Testing"
+slug: sig-testing-spotlight-2023
+date: 2023-11-24
+author: "Sandipan Panda"
+---
+
+Welcome to another edition of the SIG Spotlight blog series, where we
```
Comment (sftim) [inline content/en/blog/2023/sig-testing-spotlight.md:9]:
```suggestion
Welcome to another edition of the _SIG spotlight_ blog series, where we
```

This'll avoid readers thinking that we have a Spotlight special interest group, even momentarily.

File: content/en/blog/2023/sig-testing-spotlight.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+code cannot get updated immediately because there is so much of it, in
+particular for E2E testing. The initiative to [apply stricter linting to new or modified code](https://groups.google.com/a/kubernetes.io/g/dev/c/myGiml72IbM/m/QdO5bgQiAQAJ)
+while accepting that existing code doesn’t pass those same linter
+checks helps a bit.
+
+**Sandipan:** Any SIG accomplishments that you are proud of and would
+like to highlight?
+
+**Patrick:** I am biased because I have been driving this, but I think
+that the E2E framework and linting are now in a much better shape than
```
Comment (sftim) [inline content/en/blog/2023/sig-testing-spotlight.md:119]: Optionally, add one more link
```suggestion
that the [E2E framework](https://github.com/kubernetes-sigs/e2e-framework) and linting are now in a much better shape than
```

File: content/en/blog/2023/sig-testing-spotlight.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+tools exist; how did you arrive at the existing practices?
+
+**Patrick:** I can’t speak about the early days because I wasn’t
+around yet 😆, but looking back at some of the commit history it’s
+pretty obvious that developers just took what was available and
+started using it. For E2E testing, that was
+[Ginkgo+Gomega](https://github.com/onsi/ginkgo). Some hacks were
+necessary, for example around cleanup after a test run and for
+categorising tests. Eventually this led to Ginkgo v2 and [revised best
+practices for E2E testing](https://www.kubernetes.dev/blog/2023/04/12/e2e-testing-best-practices-reloaded/).
```
Comment (sftim) [inline content/en/blog/2023/sig-testing-spotlight.md:55]:
```suggestion
practices for E2E testing](/blog/2023/04/12/e2e-testing-best-practices-reloaded/).
```
here; change it back when mirroring the article though.

File: content/en/blog/2023/sig-testing-spotlight.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+**Patrick:** I am biased because I have been driving this, but I think
+that the E2E framework and linting are now in a much better shape than
+they used to be. We may soon be able to run integration tests with
+race detection enabled, which is important because we currently only
+have that for unit tests and those tend to be less complex.
+
+**Sandipan:** Testing is always important, but is there anything
+specific to your work in terms of the Kubernetes release process?
+
+**Patrick:** [test flakes](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/flaky-tests.md)...
```
Comment (sftim) [inline content/en/blog/2023/sig-testing-spotlight.md:127]: (nit)
```suggestion
**Patrick:** [test flakes](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/flaky-tests.md)…
```

File: content/en/blog/2023/sig-testing-spotlight.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+could tell us?
+
+**Patrick:** I started working on E2E framework enhancements five
+years ago, then was less active there for a while. When I came back
+and wanted to test some new enhancement, I asked about how to write
+unit tests for the new code and was pointed to some existing tests
+which looked vaguely familiar, as if I had *seen* them before. I
+looked at the commit history and found that I had *written* them! I’ll
+let you decide whether that says something about my failing long-term
+memory or simply is normal... Anyway folks, remember to write good
```
Comment (sftim) [inline content/en/blog/2023/sig-testing-spotlight.md:157]: (nit)
```suggestion
memory or simply is normal… Anyway, folks, remember to write good
```

File: content/en/blog/2023/sig-testing-spotlight.md [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+commit messages and comments, someone will need them at some point -
```
Comment (sftim) [inline content/en/blog/2023/sig-testing-spotlight.md:158]: (nit / opinion)
```suggestion
commit messages and comments; someone will need them at some point -
```

**Review by sftim — COMMENTED** [review]
Nits with one thing I recommend changing.

**Comment by sftim** [comment]
https://github.com/kubernetes/contributor-site/pull/454#discussion_r1398185584 is the one thing I would like to see changed; the other feedback is optional.

**Comment by sftim** [comment]
/hold

Once https://github.com/kubernetes/website/pull/43982 is approved, it's OK to unhold this.
If it turned out that https://github.com/kubernetes/website/pull/43982 didn't get reviewed in time (there's little reviewer capacity), it's also OK to skip the mirroring and unhold to publish just this article.

**Comment by fsmunoz** [comment]
Thank you @sftim , I added all your suggestions, and will add them to [43982](https://github.com/kubernetes/website/pull/43982) as well.

**Comment by fsmunoz** [comment]
Both this one and the one on `website` are on hold waiting for the other; to break the loop, I'll unhold this one first.

/unhold

**Comment by sftim** [comment]
/hold

We should get both articles LGTMed before unholding.

**Comment by sftim** [comment]
If this gets LGTM and approval, and the deadline arrives, then it's OK to unhold.
I'll take a look at both articles.

**Comment by sftim** [comment]
@fsmunoz https://github.com/kubernetes/website/pull/43982 is marked as a work in progress. Is that intentional?

**Comment by fsmunoz** [comment]
@sftim fixed there, thanks!

**Comment by sftim** [comment]
/hold cancel
https://github.com/kubernetes/website/pull/43982 is approved and LGTM

**Pull request merged**